Functions with `$INODE64` suffix not found on darwin/aarch64

Compiling on M1 fails with:

```
clang: warning: /usr/bin/clang: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
: error: /usr/bin/clang: Undefined symbols for architecture arm64:
: error: /usr/bin/clang:   "_fstat$INODE64", referenced from:
: error: /usr/bin/clang:       __Jjava_io_IO_1Util_IO_1GetLength__Ljava_io_FileDescriptor_2_J in IO_Util.o
: error: /usr/bin/clang:       __Jjava_io_IO_1Util_handleOpen__Lorg_qbicc_runtime_CNative_00024ptr_2Lorg_qbicc_runtime_CNative_00024c_1int_2Lorg_qbicc_runtime_CNative_00024c_1int_2_J in IO_Util.o
: error: /usr/bin/clang:       __Jsun_nio_ch_FileDispatcherImpl_size0__Ljava_io_FileDescriptor_2_J in FileDispatcherImpl.o
: error: /usr/bin/clang:   "_lstat$INODE64", referenced from:
: error: /usr/bin/clang:       __Jsun_nio_fs_UnixNativeDispatcher_lstat0__JLsun_nio_fs_UnixFileAttributes_2_V in UnixNativeDispatcher.o
: error: /usr/bin/clang:   "_stat$INODE64", referenced from:
: error: /usr/bin/clang:       __Jjava_io_UnixFileSystem_getBooleanAttributes0__Ljava_io_File_2_I in UnixFileSystem.o
: error: /usr/bin/clang:       __Jsun_nio_fs_UnixNativeDispatcher_stat1__J_I in UnixNativeDispatcher.o
: error: /usr/bin/clang:       __Jsun_nio_fs_UnixNativeDispatcher_stat0__JLsun_nio_fs_UnixFileAttributes_2_V in UnixNativeDispatcher.o
ld: error: /usr/bin/clang: symbol(s) not found for architecture arm64
clang: error: /usr/bin/clang: linker command failed with exit code 1 (use -v to see invocation)
error: Linker invocation failed: org.qbicc.machine.tool.CompilationFailureException: Compiler terminated with exit code 1
Compilation failed with 14 error(s) and 121 warning(s)
make: *** [Makefile:122: run] Error 1
```

The `$INODE64` suffix is not needed in darwin/aarch64 (as hinted in https://github.com/al45tair/mac_alias/issues/10). Once the suffix is removed, the code compiles fine.

For a proper fix, we probably need to adjust the `when` values to take into account both os and arch. I'll explore this.

--- a/runtime/posix/src/main/java/org/qbicc/runtime/posix/SysStat.java
+++ b/runtime/posix/src/main/java/org/qbicc/runtime/posix/SysStat.java
@@ -73,12 +73,12 @@ public final class SysStat {
     public static final mode_t S_IWOTH = constant();
     public static final mode_t S_IXOTH = constant();
 
-    @name(value = "stat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "stat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
     public static native c_int stat(const_char_ptr pathName, struct_stat_ptr statBuf);
-    @name(value = "fstat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "fstat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
     public static native c_int fstat(c_int fd, struct_stat_ptr statBuf);
-    @name(value = "lstat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "lstat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
     public static native c_int lstat(const_char_ptr pathName, struct_stat_ptr statBuf);
-    @name(value = "fstatat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "fstatat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
     public static native c_int fstatat(c_int dirFd, const_char_ptr pathName, struct_stat_ptr statBuf, c_int flags);
 }

